connector-packager/tests/test_connector_properties.py

```diff
@@ -0,0 +1,47 @@
+import unittest
+import logging
+from pathlib import Path
+
+from .jar_packager import create_jar
+from connector_packager.connector_file import ConnectorFile
+from connector_packager.connector_properties import ConnectorProperties
+from connector_packager.xsd_validator import validate_all_xml
+
+logger = logging.getLogger(__name__)
+
+TEST_FOLDER = Path("tests/test_resources")
+
+
+class TestConnectorProperties(unittest.TestCase):
+
+    def test_uses_tcd_property(self):
+        # Check that validate_all_xml properly sets uses_tcd to True if using .tcd file
+        test_folder = TEST_FOLDER / Path("valid_connector")  # This connector uses a .tcd file
+
+        files_list = [
+            ConnectorFile("manifest.xml", "manifest"),
+            ConnectorFile("connection-dialog.tcd", "connection-dialog"),
+            ConnectorFile("connectionBuilder.js", "script"),
+            ConnectorFile("dialect.tdd", "dialect"),
+            ConnectorFile("connectionResolver.tdr", "connection-resolver"),
+            ConnectorFile("resources-en_US.xml", "resource")]
+
+        properties_uses_tcd = ConnectorProperties()
+        self.assertTrue(validate_all_xml(files_list, test_folder, properties_uses_tcd), "Valid connector not marked as valid")
+        self.assertTrue(properties_uses_tcd.uses_tcd, "uses_tcd not set to True for connector using .tcd file")
+
+        # Check that validate_all_xml properly sets uses_tcd to False if not using .tcd file
+        test_folder = TEST_FOLDER / Path("modular_dialog_connector")  # This connector uses a conneciton-fields.xml file
+
+        files_list = [
```

Comment:
can you re-use/append to the above one? slightly less copy/paste

Reply:
I could maybe re-use the common files, but there's actually only 2 files in common because some of the file extensions are different, and I feel more comfortable explicitly listing the files each time to prevent test errors that are hard to diagnose.